File: EcoArt/src/product/product.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package product;

import java.util.Objects;
/**
 *
 * @author User
 */
public class product {
    private long id_pdts;
    private String nom;
    private double prix;
    private int qte ;
    private String categ;
    private String matiere;
    private String description;
    
   
    
    
    public product (long id_pdts,String nom ,double prix,int qte,String categ,String matiere,String description){
    this.id_pdts=id_pdts;
    this.nom=nom;
    this.prix=prix;
    this.qte=qte;
    this.categ=categ;
    this.matiere=matiere;
    this.description=description;
    
    }

    public product() {
    }
    

    
    public long getId_pdts() {
        return id_pdts;
    }

    public void setId_pdts(long id_pdts) {
        this.id_pdts = id_pdts;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getCateg() {
        return categ;
    }

    public void setCateg(String categ) {
        this.categ = categ;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public String getMatiere() {
        return matiere;
    }

    public void setMatiere(String matiere) {
        this.matiere = matiere;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String desc) {
        this.description = desc;
    }

    public int getQte() {
        return qte;
    }

    public void setQte(int qte) {
        this.qte = qte;
    }

   

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final product other = (product) obj;
        if (this.id_pdts != other.id_pdts) {
            return false;
        }
        if (Double.doubleToLongBits(this.prix) != Double.doubleToLongBits(other.prix)) {
            return false;
        }
        if (this.qte != other.qte) {
            return false;
        }
        if (!Objects.equals(this.nom, other.nom)) {
            return false;
        }
        if (!Objects.equals(this.categ, other.categ)) {
            return false;
        }
        if (!Objects.equals(this.matiere, other.matiere)) {
            return false;
        }
        if (!Objects.equals(this.description, other.description)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "product{" + ", nom=" + nom + ", categ=" + categ + ", prix=" + prix + ", matiere=" + matiere + ", desc=" + description + ", qte=" + qte + '}';
    }

    public product(String nom, double prix, int qte, String categ, String matiere, String description) {
        this.nom = nom;
        this.prix = prix;
        this.qte = qte;
        this.categ = categ;
        this.matiere = matiere;
        this.description = description;
    }
    

  
}

File: testweb/src/Entity/Product.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use App\Repository\ProductRepository;

#[UniqueEntity(fields: 'nom', message: 'Ce nom de produit est déjà utilisé.')]
#[ORM\Entity(repositoryClass:ProductRepository::class)]
class Product
{
  
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]

    private ? int $idpdts=null;

   
    #[Assert\NotBlank]
    #[Assert\Length(max:50)]
    #[Assert\NotBlank(message:"Name can't be empty")]
    #[ORM\Column(length:50, unique:true)]
    private ? string $nom=null;
    
    #[Assert\NotBlank(message:"price can't be empty")]
    #[ORM\Column]
    private ?float $prix=null;

    #[Assert\NotBlank(message:"Quantity can't be empty")]
    #[ORM\Column]
    private ?int $qte=null;

    #[Assert\NotBlank(message:"Category can't be empty")]
    #[ORM\Column(length:50)]
    private ?String $categ=null;

    
    #[Assert\NotBlank(message:"Metrial can't be empty")]
    #[ORM\Column(length:50)]
    private ? string $matiere=null;

    #[Assert\NotBlank(message:"Description can't be empty")]
    #[ORM\Column(length:350)]
    private ?string $description=null;


   // #[Assert\NotBlank(message:"Picture can't be empty")]
    #[ORM\Column(length:1000)]
    private ?string $image=null;


    public function getIdpdts(): ?int
    {
        return $this->idpdts;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getQte(): ?int
    {
        return $this->qte;
    }

    public function setQte(int $qte): self
    {
        $this->qte = $qte;

        return $this;
    }

    public function getCateg(): ?string
    {
        return $this->categ;
    }

    public function setCateg(string $categ): self
    {
        $this->categ = $categ;

        return $this;
    }

    public function getMatiere(): ?string
    {
        return $this->matiere;
    }

    public function setMatiere(string $matiere): self
    {
        $this->matiere = $matiere;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

  


}